proposals/0180-leader-schedule-migration.md

```diff
@@ -0,0 +1,91 @@
+---
+simd: '0180'
+title: Leader Schedule Migration
+authors: Justin Starry (Anza)
+category: Standard
+type: Core
+status: Draft
+created: 2024-10-03
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+The epoch leader schedule for block production will be migrated from using
+validator identity addresses to using vote account addresses. The expected
+block signer for a given slot will be determined by the vote account's 
+designated validator identity.
+
+## Motivation
+
+Using validator identity addresses in the leader schedule means there is no
+straightforward way to map a block producer to a particular vote account and its
+delegated stake. This is becuase the same validator identity could be designated
+by multiple vote accounts. By migrating to the vote account address, we know
+exactly what delegated stake led to a validator's leader schedule slot
+allocation. This will make certain protocol improvements much easier to design
+like how to distribute block rewards and how to slash validators that produce
+duplicate blocks.
+
+## New Terminology
+
+NA
+
+## Detailed Design
+
+### Leader Schedule Generation
+
+When generating the leader schedule at epoch boundaries, rather than
+accumulating all stake by the node id, stake should be accumulated according to
+vote pubkey. Then use the existing stake weighted randomized leader schedule
+generation using vote pubkeys and their delegated stake rather than node id
+pubkeys and the accumulated delegated stake across (potentially more than one)
+vote accounts. As before, only valid and initialized vote accounts should be
+used during leader schedule generation.
+
+### Node Id Lookup
+
+Block shreds should still be signed by a node pubkey and block rewards should
+still be collected into the node id account (also known as fee collection
+account). However, after the migration this node pubkey will need to be looked
+up by first finding the vote account for the designated vote pubkey for a
+particular leader slot in bank epoch stakes. Bank epoch stakes are keyed by
+leader schedule epoch and therefore the vote account state should be retrieved
+by looking up the epoch stakes for the current epoch. Since only valid vote
```

Comment:
Confirming my understanding that this shouldn't cause an issue at epoch boundary:
Since the leader schedule is generated an epoch before, if we receive a shred from epoch E + 1, we can still use the root bank from epoch E to perform the vote account -> node pubkey lookup.

Reply:
That's correct. We can only validate shreds for epoch E + 1 once we have a root bank in epoch E